src/Casbin.AspNetCore/ServiceCollectionExtension.cs

```diff
@@ -1,13 +1,49 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using System;
+using Casbin.AspNetCore.Abstractions;
+using Casbin.AspNetCore.Policy;
+using Casbin.AspNetCore.Transformers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using NetCasbin;
+using NetCasbin.Model;
 
 namespace Casbin.AspNetCore
 {
     public static class ServiceCollectionExtension
     {
-        public static IServiceCollection AddCasbinAuthorization(this IServiceCollection services)
+        public static IServiceCollection AddCasbinAuthorizationCore(this IServiceCollection services, Action<CasbinAuthorizationCoreOptions> configure)
         {
-
+            var options = CasbinAuthorizationCoreOptions.GetDefault();
+            configure(options);
+            CheckOptions(options);
+            services.TryAdd(ServiceDescriptor.Describe(typeof(Model),
+                p => options.ModelFactory?.Invoke(),
+                options.ModelLifeTime));
+            services.TryAdd(ServiceDescriptor.Describe(typeof(Enforcer),
+                p =>
+                    options.EnforcerFactory?.Invoke(p.GetRequiredService<Model>()),
+                options.EnforcerLifeTime));
+            services.TryAddSingleton<
+                ICasbinAuthorizationContextFactory,
+                DefaultCasbinAuthorizationContextFactory>();
+            services.TryAddScoped<IAuthorizationHandler, CasbinAuthorizationHandler>();
+            services.AddSingleton<IRequestTransformer, BasicRequestTransformer>();
```

Comment:
People may not use `name` to store the current user, for me, I use Jwt and use `user_id` to store the current user id, how can this `BasicRequestTransformer` knows if I want `name` or `user_id`?

Reply:
You can add a new `RequestTransformer` and can be set default `RequestTransformer` at the CasbinAuthorizationRequirement. You can also set the `RequestTransformer` when executing `AuthorizeAsync`.
For the default provide `RequestTransformers` at the package, we can add options about it and use a factory to create them. What do you think about it?

Reply:
How  about add a NameIdentifierKey in options? By default ,we can define a const string for common usage.

Reply:
Maybe I'm not using aspnet properly, found a question on stackoverflow about how to get current user https://stackoverflow.com/questions/36641338/how-to-get-current-user-in-asp-net-core

![image](https://user-images.githubusercontent.com/4054836/86559210-064a1100-bf8e-11ea-8b09-a231fd1664e2.png)

May have some unified methods to get current user, but I don't know :grin: